disco_aws_automation/disco_elastigroup.py

```diff
@@ -0,0 +1,242 @@
+
+    @property
+    def token(self):
+        """
+        Returns spotinst auth token from JSON file in ~/.aws/spotinst_api_token
+
+        File format example:
+
+        {
+          "name": "user_ampli
+          "token": "f7e6c5abb51bb04fcaa411b7b70cce414c821bf719f7db0679b296e588630515"
+        }
+        """
+        token_file = json.load(open(expanduser('~') + '/.aws/spotinst_api_token'))
+        if token_file:
+            self._token = token_file['token']
+        return self._token
+
+    @property
+    def session(self):
+        """Lazily create session object"""
+        if not self._session:
+            self._session = requests.Session()
+
+        # Insert auth token in header
+        self._session.headers.update(
+            {
+                "Content-Type": "application/json",
+                "Authorization": "Bearer {}".format(self.token)
+            }
+        )
+        return self._session
+
+    def get_new_groupname(self, hostclass):
+        """Returns a new elastigroup name when given a hostclass"""
+        return self.environment_name + '_' + hostclass + "_" + str(int(time.time()))
+
+    def _filter_by_environment(self, groups):
+        """Filters elastigroups by environment"""
+        return [
+            group for group in groups
+            if group['name'].startswith("{0}_".format(self.environment_name))
+        ]
+
+    def get_hostclass(self, group_name):
+        """Returns the hostclass when given an elastigroup name"""
+        return group_name.split('_')[1]
+
+    def get_existing_groups(self, hostclass=None, group_name=None):
+        """
+        Returns all elastigroups for a given hostclass, sorted by most recent creation. If no
+        elastigroup can be found, returns an empty list.
+        """
+        try:
+            groups = self.session.get(SPOTINST_API).json()['response']['items']
+        except KeyError:
+            return []
+        filtered_groups = [group for group in groups
+                           if not hostclass or self.get_hostclass(group_name) == hostclass]
+        filtered_groups.sort(key=lambda group: group["name"], reverse=True)
+        return filtered_groups
+
+    def get_group_ids(self, hostclass=None, group_name=None):
+        """Returns list of elastigroup ids filtered by hostclass or group_name"""
+        groups = self.get_existing_groups(hostclass, group_name)
+        group_ids = [group["id"] for group in groups]
+        return group_ids
+
+    def get_group_instances(self, group_id):
+        """Returns list of instance ids in a group"""
+        instances = self.session.get(SPOTINST_API + group_id + '/status').json()['response']['items']
+        return [instance['instanceId'] for instance in instances]
+
+    def create_elastigroup_config(self, hostclass, availability_vs_cost, desired_size, min_size, max_size,
+                                  instance_type, zones, load_balancers, security_groups, instance_monitoring,
```

Comment:
Should the param``` instance_type``` be rename ```instance_types``` as it includes a comma separated list of instance types?